firedrake/mg/utils.py

```diff
@@ -140,6 +140,8 @@ def physical_node_locations(V):
         return cache[key]
     except KeyError:
         Vc = firedrake.FunctionSpace(mesh, ufl.VectorElement(element))
+        locations = firedrake.Function(Vc)
```

Comment:
Maybe a dumb question: doesn't this create a variable, then immediately overwrite it?

Reply:
Oh yeah!

Reply:
Fixed.